newspaper/images.py

```diff
@@ -167,6 +171,52 @@ def fetch_image_dimension(url, useragent, referer=None, retries=1):
     return fetch_url(url, useragent, referer, retries, dimension=True)
 
 
+def valid_image_url(url):
+
+    url = clean_url(url)
+    if not url.startswith(('http://', 'https://')):
+        return False
+    path = urlparse(url).path
+
+    # input url is not in valid form (scheme, netloc, tld)
+    if not path.startswith('/'):
+        return False
+
+    # the '/' which may exist at the end of the url provides us no information
+    if path.endswith('/'):
+        path = path[:-1]
+
+    # '/story/cnn/blahblah/index.html' --> ['story', 'cnn', 'blahblah', 'index.html']
+    path_chunks = [x for x in path.split('/') if len(x) > 0]
+
+    # siphon out the file type. eg: jpeg, png
+    if len(path_chunks) > 0:
+        file_type = urls.url_to_filetype(url)
+
+        # if the file type is a media type, reject instantly
+        if file_type and file_type not in ALLOWED_TYPES:
```

Comment:
This is good because if the file_type is "" -> if the URL resource does not specify the file suffix we also don't aggressively reject the file.